tools/commands-cordova.js

```diff
@@ -638,38 +673,64 @@ var ensureCordovaPlugins = function (projectContext, options) {
     }
   });
 
-  if (shouldReinstallPlugins) {
+  if (! _.isEmpty(pluginsFromLocalPath))
+    verboseLog('Reinstalling cordova plugins added from the local path');
+
+  if (shouldReinstallPlugins || ! _.isEmpty(pluginsFromLocalPath)) {
     // Loop through all of the current plugins and remove them one by one until
-    // we have no plugins. It's necessary to loop because we might have
-    // dependencies between plugins.
-    var uninstallAllPlugins = function () {
+    // we have deleted proper amount of plugins. It's necessary to loop because
```

Comment:
Any reason why we would uninstall only some plugins and not all of them? Previously we noticed that failures on uninstalling a couple of plugins and installing the rest are pretty often and a "safe" way would be uninstalling everything at once.